controllers/configurationpolicy_utils.go

```diff
@@ -307,14 +307,19 @@ func formatMetadata(metadata map[string]interface{}) (formatted map[string]inter
 		md["labels"] = labels
 	}
 	if annos, ok := metadata["annotations"]; ok {
-		noAutogenerated := map[string]interface{}{}
-		for key := range annos.(map[string]interface{}) {
-			if !isAutogenerated(key) {
-				noAutogenerated[key] = annos.(map[string]interface{})[key]
+		// When a non-map is provided, skip iterating and set the value directly
+		if iterableAnnos, ok := annos.(map[string]interface{}); ok {
```

Comment:
Could you consolidate the two if statements?

For example:
```go
	if annos, ok := metadata["annotations"].(map[string]interface{}); ok {
```

This will work because if the annotations key is not set, then an `interface{}` value is returned. If it is set, then the type assertion will only work if it's actually valid.

Reply:
Ah, yes. I was trying to poke at how to combine them. Thanks!

Reply:
Actually, in order to preserve the `metadata.annotations` actual value, we do need two if statements...

Reply:
@dhaiducek ah, I see. How about this then?

```go
	if annos, ok := metadata["annotations"].(map[string]interface{}); ok {
		....
	} else if annos, ok := metadata["annotations"]; ok {
		md["annotations"] = annos
	}
```